src/middlewared/middlewared/plugins/audit/backend.py

```diff
@@ -126,9 +126,12 @@ def setup(self):
         - after audit database deletion or rename
         """
         for svc, conn in self.connections.items():
+            # Dismiss any existing AuditSetup one-shot alerts
+            self.middleware.call_sync('alert.oneshot_delete', 'AuditBackendSetup', {"service": svc})
             try:
                 conn.setup()
             except Exception:
+                self.middleware.call_sync('alert.oneshot_create', 'AuditBackendSetup', {"service": svc})
```

Comment:
I think maybe we should build list of failures during setup (because there can be more than one) and generate / delete the alert before returning.